pkg/account/delete/client.go

```diff
@@ -68,6 +71,79 @@ func (c *AccountAPIClient) DeleteUser(ctx context.Context, email string) error {
 	return nil
 }
 
+// DeleteServiceUser retrieves all service users, looks up the service user by name and removes it from the account.
+// Returns error if any API call fails unless the user is already not present, either in the list when looking by name or during the subsequent delete call.
+// In addition, an error is returned if multiple service users are found with the same name.
+func (c *AccountAPIClient) DeleteServiceUser(ctx context.Context, name string) error {
+	uid, err := c.getServiceUserIDByName(ctx, c.accountUUID, name)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.client.ServiceUserManagementAPI.DeleteUser(ctx, c.accountUUID, uid).Execute()
+	defer closeResponseBody(resp)
+	if resp != nil && resp.StatusCode == 404 {
+		return NotFoundErr
+	}
+	if err := handleClientResponseError(resp, err, fmt.Sprintf("failed to delete service user %q", name)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *AccountAPIClient) getServiceUserIDByName(ctx context.Context, accountUUID, name string) (string, error) {
+	serviceUsers, err := c.getServiceUsers(ctx, accountUUID)
+	if err != nil {
+		return "", err
+	}
+
+	uid := ""
+	for _, s := range serviceUsers {
+		if s.Name == name {
+			if uid != "" {
+				return "", fmt.Errorf("found multiple service users with name %s", name)
+			}
+			uid = s.Uid
+		}
+	}
+	if uid == "" {
+		return "", NotFoundErr
+	}
+
+	return uid, nil
+}
+
+func (c *AccountAPIClient) getServiceUsers(ctx context.Context, accountUUID string) ([]accountmanagement.ExternalServiceUserDto, error) {
+	serviceUsers := []accountmanagement.ExternalServiceUserDto{}
+	const pageSize = 100
+	for page := (int32)(1); page < math.MaxInt32; page++ {
```

Comment:
I don't see the benefit of this extra caution.
In case the Dynatrace is broken, it's more likely that the anti-DoS guardian will activate and the request will fail.
Even when the whole loop succeeds, and for loop bump in the upper limit, this code doesn't return an error. Instead, it pretends everything is OK, giving a false positive appearance.

On the other hand, it brings unwanted complexity to the code. And make it harder to understand.